ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/EstablishedPeers.hs

```diff
@@ -419,8 +422,16 @@ jobPromoteColdPeer PeerSelectionActions {
         let establishedPeers' = EstablishedPeers.insert peeraddr peerconn
                                                         (addTime policyPeerShareActivationDelay now)
                                                         establishedPeers
+            advertise = case peerSharing of
+                          PeerSharingEnabled  -> DoAdvertisePeer
+                          PeerSharingDisabled -> DoNotAdvertisePeer
             -- Update PeerSharing value in KnownPeers
-            knownPeers'       = KnownPeers.insert (Map.singleton peeraddr (Just peerSharing, Nothing, Nothing))
+            knownPeers'       = KnownPeers.insert (Map.singleton peeraddr ( Just peerSharing
+                                                                          , newDefaultValue peeraddr
+                                                                                            advertise
+                                                                                            knownPeers
+                                                                          , Nothing))
```

Comment:
we do two `KnownPeers` lookup here, one to `insert` another in `newDefaultValue`, why not implement `KnownPeers.alter` similar to [`Map.alter`](https://hackage.haskell.org/package/containers-0.7/docs/Data-Map-Internal.html#v:alter).